packages/language/src/linking/resolver.ts

```diff
@@ -232,7 +232,11 @@ function getMatchingSymbols(
   qualifiedName: string[],
   reference: Reference,
   reporter: LinkerErrorReporter,
-): QualifiedSyntaxNode[] {
+): readonly QualifiedSyntaxNode[] {
+  if (reference.type === ReferenceType.Type) {
```

Comment:
This is a very short branch. Looking at the non-type branch: Why is it not possible to have a duplicated name / ambiguous symbols in the type branch?